### ISSUE TITLE
Calling functions with Protocols that involve operant invoked functions fail

### ISSUE THREAD

**Issue by PartiallyTyped** [python/mypy#9305]
* Are you reporting a bug, or opening a feature request? Bug

```python

T = TypeVar("T")


class ClosedUnderFormat(Protocol[T]):
    def format(self: "ClosedUnderFormat"[T], other: "ClosedUnderFormat"[T]) -> "ClosedUnderFormat[T]":
        pass


class ClosedUnderAddition(Protocol[T]):
    def __add__(self: "ClosedUnderAddition"[T], other: "ClosedUnderAddition"[T]) -> "ClosedUnderAddition[T]":
        pass


class Bar(int):
    def format(self: int, other: int):
        return ""


def foofmt(x: ClosedUnderFormat[T], y: ClosedUnderFormat[T]) -> ClosedUnderFormat[T]:
    return x.format(y)


def fooadd(x: ClosedUnderAddition[T], y: ClosedUnderAddition[T]) -> ClosedUnderAddition[T]:
    return x.__add__(y)


def fooadd_operant(x: ClosedUnderAddition[T], y: ClosedUnderAddition[T]) -> ClosedUnderAddition[T]:
    return x + y


foofmt("foo", "foo") # ok
foofmt("foo", 10) # true positive - caught 
foofmt(Bar(10), Bar(11)) # true positive - caught

fooadd("foo", "foo") # False positive - caught but shouldn't
fooadd(10, 10) # False positive - caught but shouldn't
fooadd_operant(10, 10) # False positive - caught but shouldn't


```

* What is the actual behavior/output?
```
pythoid/definitions.py:40: error: Argument 2 to "foofmt" has incompatible type "int"; expected "ClosedUnderFormat[<nothing>]" <-- Correct
pythoid/definitions.py:41: error: Argument 1 to "foofmt" has incompatible type "Bar"; expected "ClosedUnderFormat[<nothing>]" <-- Correct
pythoid/definitions.py:41: error: Argument 2 to "foofmt" has incompatible type "Bar"; expected "ClosedUnderFormat[<nothing>]" <-- Correct
pythoid/definitions.py:43: error: Argument 1 to "fooadd" has incompatible type "str"; expected "ClosedUnderAddition[<nothing>]" <-- Wrong, str implements .__add__
pythoid/definitions.py:43: error: Argument 2 to "fooadd" has incompatible type "str"; expected "ClosedUnderAddition[<nothing>]" <-- Wrong, str implements .__add__
pythoid/definitions.py:44: error: Argument 1 to "fooadd" has incompatible type "int"; expected "ClosedUnderAddition[<nothing>]" <-- Wrong, int implements .__add__
pythoid/definitions.py:44: error: Argument 2 to "fooadd" has incompatible type "int"; expected "ClosedUnderAddition[<nothing>]" <-- Wrong, int implements .__add__
pythoid/definitions.py:45: error: Argument 1 to "fooadd_operant" has incompatible type "int"; expected "ClosedUnderAddition[<nothing>]" <-- Wrong, int implements .__add__, invoked by +
pythoid/definitions.py:45: error: Argument 2 to "fooadd_operant" has incompatible type "int"; expected "ClosedUnderAddition[<nothing>]" <-- Wrong, int implements .__add__, invoked by +
```

* What is the behavior/output you expect?

```
pythoid/definitions.py:40: error: Argument 2 to "foofmt" has incompatible type "int"; expected "ClosedUnderFormat[<nothing>]" <-- Correct
pythoid/definitions.py:41: error: Argument 1 to "foofmt" has incompatible type "Bar"; expected "ClosedUnderFormat[<nothing>]" <-- Correct
pythoid/definitions.py:41: error: Argument 2 to "foofmt" has incompatible type "Bar"; expected "ClosedUnderFormat[<nothing>]" <-- Correct
```
* What are the versions of mypy and Python you are using? 
  * mypy0.782
  * Python 3.8.3

* Do you see the same issue after installing mypy from Git master? Yes
* What are the mypy flags you are using? None

**Comment by JukkaL** [comment]
Yeah, this seems inconsistent.

**Comment by ilevkivskyi** [comment]
I think this is a duplicate of https://github.com/python/mypy/issues/3829. Btw thank for the real life example!
